### PR TITLE
[WIP] func_instance continuation and some lighting preview improvements

### LINKED ISSUE #4: Lighting preview hides models upon selection and sprites
Caused due to [a check that prevents things that should be rendered last to be added to rendering](https://github.com/Gocnak/sdk-2013-hammer/blob/master/hammer/Render3DMS.cpp#L2313) in Lighting Preview modes.

### LINKED ISSUE #5: Re-add func_instance support
SCell already started this, but still needs:

- [ ] Tint entities inside instance to orange
- [ ] Disable selection of entities inside instance
- [ ] File select dialog for instances

### LINKED ISSUE #28: Move all settings to KeyValues file
Hammer currently uses Windows registry to store everything and will need to move over to KeyValues to be able to support other platforms.

A lot of settings use `APP()->Get*`. This falls through to AFX stuff that uses Windows registry.

### PR DESCRIPTION
**Pull request by SCell555**
## Func_instance support:
- [x] Tint entities inside instance, 
- [x] Tint brushes inside instance,
- [x] Prevent selecting entities inside instance, all selects are redirected to func_instance itself
- [x] Rotating entities inside instance correctly
- [x] Prevent movement of entities inside instance after transforming selection box
- [x] Maybe prevent rendering of helpers for entities inside instance
- [x] Advanced controls on main panel
- [ ] Adjust tinting color
- [ ] Instance collapsing

## Minor changes:
- Model browser now runs idle at 30 fps, and sets itself to 60 fps when user is interacting with it. Feels much better than old 12.5 fps
- In ray-traced lighting preview, models cast shadows, triangles are taken from mesh of an entity
hammer_2018-09-06_03-34-43

## Issue Closing:
closes #4 
closes #5 
closes #28

### CODE REVIEW

File: hammer/Render3DMS.cpp [hunk 1]
```diff
@@ -2262,9 +2263,9 @@ void CRender3D::RenderMapClass(CMapClass *pMapClass)
 				}
 				else
 				{
-					if (
-						(m_eCurrentRenderMode != RENDER_MODE_LIGHT_PREVIEW2) &&
-						(m_eCurrentRenderMode != RENDER_MODE_LIGHT_PREVIEW_RAYTRACED) )
+					//if (
+					//	(m_eCurrentRenderMode != RENDER_MODE_LIGHT_PREVIEW2) &&
```
Comment (Gocnak) [inline hammer/Render3DMS.cpp]: Very nice, closes #17 when merged in

**Comment by Gocnak** [comment]
Fantastic work so far!

I can add you as a collaborator to this repo after this is merged in so you can directly branch on this repo as opposed to updating a branch on yours, if you'd like.

**Comment by Gocnak** [comment]
Just saw your edit message, I can merge it in if you'd like!

That way you can work on the instance things under the instance branch. All up to you, sorry about the delay in catching that.

**Comment by SCell555** [comment]
No problem. I would wait with merging until I finish instance collapsing. So at least it is feature-complete.

**Comment by Gocnak** [comment]
Hey, did you still need help on instance collapsing? I can try to poke at it if you're stumped or busy, if you'd like.

**Comment by SCell555** [comment]
Sorry for not getting to you sooner, I was busy last week. I'll try to get it to at least semi working state. Then we can merge it.

**Comment by SCell555** [comment]
Whoops, I totally forgot about this, I had some health issues and this slipped out of my mind.

**Comment by Gocnak** [comment]
It's all good! Would you be able to rebase on top of master? I'll get it merged in after that.

**Pull request merged**